packages/block-editor/src/components/inserter/search-items.js

```diff
@@ -116,39 +127,86 @@ export const searchItems = ( items = [], searchTerm = '', config = {} ) => {
 		return items;
 	}
 
-	const defaultGetTitle = ( item ) => item.title;
-	const defaultGetKeywords = ( item ) => item.keywords || [];
-	const defaultGetCategory = ( item ) => item.category;
+	const rankedItems = items
+		.map( ( item ) => {
+			return [ item, getItemSearchRank( item, searchTerm, config ) ];
+		} )
+		.filter( ( [ , rank ] ) => rank > 0 );
+
+	rankedItems.sort( ( [ , rank1 ], [ , rank2 ] ) => rank2 - rank1 );
+	return rankedItems.map( ( [ item ] ) => item );
+};
+
+/**
+ * Get the search rank for a given item and a specific search term.
+ * The higher is higher for items with the best match.
+ * If the rank equals 0, it should be excluded from the results.
+ *
+ * @param {Object} item       Item to filter.
+ * @param {string} searchTerm Search term.
+ * @param {Object} config     Search Config.
+ * @return {number}           Search Rank.
+ */
+export function getItemSearchRank( item, searchTerm, config = {} ) {
+	const defaultGetName = ( it ) => it.name || '';
+	const defaultGetTitle = ( it ) => it.title;
+	const defaultGetKeywords = ( it ) => it.keywords || [];
+	const defaultGetCategory = ( it ) => it.category;
 	const defaultGetCollection = () => null;
 	const defaultGetVariations = () => [];
+
 	const {
+		getName = defaultGetName,
 		getTitle = defaultGetTitle,
 		getKeywords = defaultGetKeywords,
 		getCategory = defaultGetCategory,
 		getCollection = defaultGetCollection,
 		getVariations = defaultGetVariations,
 	} = config;
 
-	return items.filter( ( item ) => {
-		const title = getTitle( item );
-		const keywords = getKeywords( item );
-		const category = getCategory( item );
-		const collection = getCollection( item );
-		const variations = getVariations( item );
+	const name = getName( item );
+	const title = getTitle( item );
+	const keywords = getKeywords( item );
+	const category = getCategory( item );
+	const collection = getCollection( item );
+	const variations = getVariations( item );
+
+	const sanitizedSearchTerm = sanitizeTerm( searchTerm );
```

Comment:
Should we pass this in `config` and check if exists there first and have the call as a fallback, to avoid calling the same function, since we do call it in `searchItems` before calling `getItemSearchRank`?

Reply:
I'm not sure I understand here.

Reply:
It was just a micro optimization to pass `normalizedSearchTerms` through `config` param because it was being calculated outside and the result is always the same.

So in `searchItems` we would add `config.normalizedSearchTerms = normalizedSearchTerms` and then in the `getItemSearchRank` function we could have something like this

```
const sanitizedSearchTerm = config.normalizedSearchTerms || sanitizeTerm( searchTerm );
```

But small impact since it's a small sync function..

Reply:
Wrong line ... sorry about that..
I meant below about this line
`const normalizedSearchTerms = words( sanitizedSearchTerm );`

Reply:
Oh I got, you're right we have a duplication but I don't think changing the meaning of the arguments is a good way to solve the problem. I think if it really becomes a performance issue, adding shortcuts and denormalization like that helps but right now I believe it just makes the semantics of the functions more confusing for no measurable impact.